File: app/storage/file/llm_config_store.py
"""LLM Provider 配置文件存储。

布局：data/llm_configs/{name}.json
注意：api_key 以明文存储，仅适合本地开发环境。
"""

from __future__ import annotations

from pathlib import Path
from typing import Any

from app.config.settings import get_settings
from app.storage.file.base import write_json_atomic, read_json, list_json_ids


class LLMConfigStore:
    """将 LLMProviderConfig 序列化为 data/llm_configs/{name}.json。"""

    def _path(self, name: str) -> Path:
        # 文件名用 name，需替换不安全字符
        safe_name = name.replace("/", "_").replace("\\", "_")
        return get_settings().data_dir / "llm_configs" / f"{safe_name}.json"

    def save(self, data: dict[str, Any]) -> None:
        write_json_atomic(self._path(data["name"]), data)

    def get(self, name: str) -> dict[str, Any] | None:
        return read_json(self._path(name))

    def delete(self, name: str) -> bool:
        path = self._path(name)
        if path.exists():
            path.unlink()
            return True
        return False

    def list_all(self) -> list[dict[str, Any]]:
        """读取全部已保存的 LLM 配置。"""
        ids = list_json_ids(get_settings().data_dir / "llm_configs")
        result = []
        for name in ids:
            data = self.get(name)
            if data:
                result.append(data)
        return result

File: miniagents-assembled/src/main/java/com/codex/miniagents/infrastructure/storage/file/LLMConfigStore.java
package com.codex.miniagents.infrastructure.storage.file;

import com.codex.miniagents.config.MiniAgentsProperties;
import com.codex.miniagents.llm.model.LlmProviderConfig;

import org.springframework.stereotype.Repository;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Repository
public class LLMConfigStore {
    private final Path baseDir;

    private final FileJsonSupport jsonSupport;

    public LLMConfigStore(MiniAgentsProperties properties, FileJsonSupport jsonSupport) {
        this.baseDir = properties.getDataDir().resolve("llm_configs");
        this.jsonSupport = jsonSupport;
    }

    private Path path(String name) {
        String safe = name.replace("/", "_").replace("\\", "_");
        return baseDir.resolve(safe + ".json");
    }

    public void save(LlmProviderConfig config) {
        jsonSupport.writeJsonAtomic(path(config.getName()), config);
    }

    public Optional<LlmProviderConfig> get(String name) {
        return jsonSupport.readJson(path(name), LlmProviderConfig.class);
    }

    public boolean delete(String name) {
        Path p = path(name);
        try {
            return Files.deleteIfExists(p);
        } catch (IOException e) {
            throw new RuntimeException("Failed to delete LLM config: " + name, e);
        }
    }

    public List<LlmProviderConfig> listAll() {
        List<LlmProviderConfig> results = new ArrayList<>();
        for (String id : jsonSupport.listJsonIds(baseDir)) {
            get(id).ifPresent(results::add);
        }
        return results;
    }
}
